main/src/ca/uwaterloo/flix/api/lsp/provider/completion/Completion.scala

```diff
@@ -316,11 +334,35 @@ object Completion {
   /**
     * Represents a keyword completion.
     *
-    * @param name the name of the keyword.
-    * @param priority the completion priority of the keyword.
+    * @param name      the name of the keyword.
+    * @param priority  the completion priority of the keyword.
     */
   case class KeywordCompletion(name: String, priority: Priority) extends Completion
 
+  /**
+    * Represents a keyword literal completion (i.e. `true`)
+    *
+    * @param name      the name of the keyword.
+    * @param priority  the priority of the keyword.
+    */
+  case class KeywordLiteralCompletion(name: String, priority: Priority) extends Completion
+
+  /**
+    * Represents a completion for a keyword with call syntax (i.e. `checked_cast`)
+    * 
+    * @param name      the name of the keyword.
+    * @param priority  the priority of the keyword.
+    */
+  case class KeywordCallCompletion(name: String, priority: Priority) extends Completion
```

Comment:
In general the checked_cast and all its friends are really naughty. 

Ditto for the debugging facilities, but perhaps less so.

I am leaning towards leaving them out so as not to misguide the programmer.

Reply:
Another question: If we were to leave them in (which I don't think we should) then the casts should probably be snippets?

Reply:
Do you mean that they don't belong in keyword completer or that they text type for the completionitem should be a snippet type?

Either way, I'd feel fine about removing them. I tried as best I could to figure out the different keywords and while I did leave a couple out, I took a very conservative approach. I figured it would be better to include too much and then go remove some stuff after feedback :)

Reply:
Drop this.